rio/src/RIO/Process.hs

```diff
@@ -672,9 +673,21 @@ augmentPath dirs mpath =
 augmentPathMap :: [FilePath] -> EnvVars -> Either ProcessException EnvVars
 augmentPathMap dirs origEnv =
   do path <- augmentPath dirs mpath
-     return $ Map.insert "PATH" path origEnv
+     return $ Map.insert "PATH" path normEnv
   where
-    mpath = Map.lookup "PATH" origEnv
+    -- On windows, env vars are case insensitive.
+    -- We pick the "last" PATH variable and ensure
+    -- it's full uppercase.
+    -- See https://github.com/commercialhaskell/rio/issues/234
+    normEnv 
+      | isWindows = Map.foldrWithKey
```

Comment:
This is a large change in the complexity of the operation. I'd rather see this addressed at the ingestion period, as I mentioned in the issue.

Reply:
I only see two ways to do this earlier:

1. create a smart constructor for `EnvVars` (breaking API change)
2. fix `getEnvironment` and friends in `base` (I'm not even sure that's a good idea)

For my use case `EnvVars` is constructed like so:

```hs
    curEnv <- Map.fromList . map (T.pack *** T.pack) <$> liftIO getEnvironment
```

Since the type alias is exposed, we can assume similar uses in the wild.